app/models/like.rb

```diff
@@ -0,0 +1,72 @@
+
+class Like < PostAction
+
+  self.table_name = :likes
+
+  def self.pa_type
+    PostActionType.like
+  end
+
+  def pa_type
+    self.class.pa_type
+  end
+
+  # === BEGIN CRAZY SUBCLASSING MADNESS === #
```

Comment:
Firstly I don't think we need to comment this way. I don't think it adds much to the code to just call it crazy. Either explain why something is done or let's leave out the comment altogether.

Secondly, can't this come from PostAction? It seems for Bookmark a lot of the code is shared. If we can't inherit it because it's class methods, how about using a mixin to avoid repeating ourselves?